docs/standard/native-interop/com-wrappers-source-generation.md

```diff
@@ -0,0 +1,166 @@
+---
+title: COM wrappers source generation
+description: Learn about compile-time source generation for COM wrappers in .NET.
+ms.date: 08/25/2023
+---
+
+# Source Generation for COM Wrappers
+
+.Net 8 introduces a source generator that creates an implementation of the [COM wrappers](./com-wrappers.md) API for you. The generator recognizes the <xref:System.Runtime.InteropServices.GeneratedComInterfaceAttribute>.
+
+The .Net runtime's built-in (not source generated) COM interop system generates an IL stub&mdash;a stream of IL instructions that is JIT-ed&mdash;at run time to facilitate the transition from managed code to COM, and vice-versa, on Windows only. Since this IL stub is generated at run time, it isn't compatible with [NativeAOT](../../core/deploying/native-aot/index.md), or [IL trimming](../../core/deploying/trimming/trim-self-contained.md). Generating the IL stub also takes CPU time, and can be difficult to debug when issues arise.
```

Comment:
```suggestion
The .Net runtime's built-in (not source generated) COM interop system, Windows-only, generates an IL stub&mdash;a stream of IL instructions that is JIT-ed&mdash;at run time to facilitate the transition from managed code to COM, and vice-versa. Since this IL stub is generated at run time, it isn't compatible with [NativeAOT](../../core/deploying/native-aot/index.md), or [IL trimming](../../core/deploying/trimming/trim-self-contained.md). Run time stub generation also makes diagnosing issues difficult.
```